cmd/metal-api/internal/service/partition-service.go

```diff
@@ -205,6 +205,16 @@ func (r *partitionResource) createPartition(request *restful.Request, response *
 		commandLine = *requestPayload.PartitionBootConfiguration.CommandLine
 	}
 
+	var dnsServers metal.DNSServers
+	if len(requestPayload.DNSServers) > 0 {
+		dnsServers = requestPayload.DNSServers
+	}
+
+	var ntpServers metal.NTPServers
+	if len(requestPayload.NTPServers) > 0 {
+		ntpServers = requestPayload.NTPServers
+	}
```

Comment:
Same validations as in the machine spec should be implemented here. Otherwise machines might default to the partition defaults, which are invalid.